Does serverless domain manager work with serverless framework 4?

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

## Bug Report

**Error Description**
Unable to use serverless-domain-manager with serverless framework 4. Issue when trying to NPM install

**Command Run**
npm install serverless-domain-manager --save-dev

**Console Output**
```sh
npm install serverless-domain-manager --save-dev
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: auth@1.0.0
npm error Found: serverless@4.0.35
npm error node_modules/serverless
npm error   dev serverless@"^4.0.35" from the root project
npm error
npm error Could not resolve dependency:
npm error peer serverless@"^2.60 || ^3.0.0" from serverless-domain-manager@7.3.8
npm error node_modules/serverless-domain-manager
npm error   dev serverless-domain-manager@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
```

**Domain Manager Configuration**
Not even configured yet

**Versions**
- Domain Manager version(s): latest
- Node/npm version: 20
- Serverless Version: 4.0.36
- Lambda Code: Dotnet8

--- a/package.json
+++ b/package.json
@@ -90,6 +90,6 @@
     "proxy-agent": "^6.4.0"
   },
   "peerDependencies": {
-    "serverless": "^2.60 || ^3.0.0"
+    "serverless": "^2.60 || ^3.0.0 || ^4.0.0"
   }
 }

--- a/package.json
+++ b/package.json
@@ -90,6 +90,6 @@
     "proxy-agent": "^6.4.0"
   },
   "peerDependencies": {
-    "serverless": "^2.60 || ^3.0.0 || ^4.0.0"
+    "serverless": ">=2.60"
   }
 }

